mex/common/settings.py

```diff
@@ -227,3 +216,27 @@ def resolve_paths(self) -> "BaseSettings":
             elif isinstance(value, WorkPath) and value.is_relative():
                 setattr(self, name, self.work_dir.resolve() / value)
         return self
+
+    @model_validator(mode="after")
+    def sync_settings(self) -> Self:
```

Comment:
I do not understand how this is called when calling the `get()` method. `sync_settings` is only executed upon class instantiation and `get()` does not necessarily instantiate an new object, does it? 
Example: 2 objects are instantiated and in the settingsContext. I update one of them and afterwards call `get()` on either of the two. I would assume that `sync_settings` is not called. Am I mistaken?

Reply:
we have [validate_assignment](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.validate_assignment) enabled for settings. that would call `sync_settings`